src/rust/engine/src/nodes/mod.rs

```diff
@@ -93,15 +95,25 @@ pub type NodeResult<T> = Result<T, Failure>;
 #[async_trait]
 impl Vfs<Failure> for Context {
     async fn read_link(&self, link: &Link) -> Result<PathBuf, Failure> {
-        Ok(self.get(ReadLink(link.clone())).await?.0)
+        let subject_path =
+            SubjectPath::Workspace(RelativePath::new(&link.path).map_err(|e| Self::mk_error(&e))?);
```

Comment:
There's quite a few instances of this construction. What you think about a `SubjectPath::new_workspace(...)` wrapper that can take `P: AsRef<Path>`?

Reply:
Done.